applications/DFG.prm

```diff
@@ -104,18 +104,18 @@ subsection Refinement control parameters
   set Minimum number of levels               = 0
   set Number of initial adaptive refinements = 0
   set Number of initial boundary refinements = 1
-  set Number of initial global refinements   = 2
+  set Number of initial global refinements   = 3
 end
 
 
 subsection Time stepping parameters
   set Adaptive time stepping        = false
   set Adaptive timestepping barrier = 2
   set Final time                    = 3000.0
-  set Initial time step             = 2.5e-2
-  set Maximum number of time steps  = 10
-  set Maximum time step             = 5e-2
-  set Minimum time step             = 1e-2
+  set Initial time step             = 5e-3
+  set Maximum number of time steps  = 100
+  set Maximum time step             = 5e-3
+  set Minimum time step             = 1e-3
```

Comment:
I was able to reproduce the error / issue: These lines were somehow changed which caused the solver of the diffusion step to fail in the DFG benchmark, see below.
```
Number of active cells                = 8340
Number of velocity degrees of freedom = 70528
Number of pressure degrees of freedom = 9056
Number of total degrees of freedom    = 79584

+---------+------------+------------------------------------------+
| Boundary conditions of the Velocity entity                      |
+---------+------------+------------------------------------------+
| Bdy. id |    Type    |              Function                    |
| 0       | Dirichlet  | DFG::VelocityInflowBoundaryCondition<2>  |
| 2       | Dirichlet  | ZeroFunction<2, double>                  |
| 3       | Dirichlet  | ZeroFunction<2, double>                  |
| Unconstrained boundary ids: 1                                   |
+---------+------------+------------------------------------------+
+---------+------------+------------------------------------------+
| Boundary conditions of the Pressure entity                      |
+---------+------------+------------------------------------------+
| Bdy. id |    Type    |              Function                    |
| 1       | Dirichlet  | ZeroFunction<2, double>                  |
| Unconstrained boundary ids: 0, 2, 3                             |
+---------+------------+------------------------------------------+
Solving until t = 350...
Step =      0, Current time = 0.0e+00, Next time step = 2.5e-02
+---------+------------+------------------------------------------+
| Boundary conditions of the Phi entity                           |
+---------+------------+------------------------------------------+
| Bdy. id |    Type    |              Function                    |
| 0       | Neumann    | ZeroFunction<2, double>                  |
| 2       | Neumann    | ZeroFunction<2, double>                  |
| 3       | Neumann    | ZeroFunction<2, double>                  |
| 1       | Dirichlet  | ZeroFunction<2, double>                  |
+---------+------------+------------------------------------------+
Step =      1, Current time = 2.5e-02, Next time step = 2.5e-02
    dp = +8.473130e+01 C_d = +1.304930e+02 C_l = -2.044101e-01
Step =      2, Current time = +5.000000e-02, Next time step = +2.500000e-02
    dp = +5.071253e+01 C_d = +8.949128e+01 C_l = -1.806466e-01


----------------------------------------------------
Exception in the solve method of the diffusion step: 

--------------------------------------------------------
An error occurred in line <146> of file </home/sg/Downloads/build/dealii-9.2.0/source/lac/petsc_solver.cc> in function
    void dealii::PETScWrappers::SolverBase::solve(const dealii::PETScWrappers::MatrixBase&, dealii::PETScWrappers::VectorBase&, const dealii::PETScWrappers::VectorBase&, const dealii::PETScWrappers::PreconditionerBase&)
The violated condition was: 
    false
Additional information: 
Iterative method reported convergence failure in step 200. The residual in the last step was 26.8619.

This error message can indicate that you have simply not allowed a sufficiently large number of iterations for your iterative solver to converge. This often happens when you increase the size of your problem. In such cases, the last residual will likely still be very small, and you can make the error go away by increasing the allowed number of iterations when setting up the SolverControl object that determines the maximal number of iterations you allow.

The other situation where this error may occur is when your matrix is not invertible (e.g., your matrix has a null-space), or if you try to apply the wrong solver to a matrix (e.g., using CG for a matrix that is not symmetric or not positive definite). In these cases, the residual in the last iteration is likely going to be large.
--------------------------------------------------------

Aborting!
----------------------------------------------------
```
This happens for Trilinos and PETSc.

Reply:
Using `dealii::SolverGMRES<LinearAlgebra::MPI::Vector>` instead of these lines
https://github.com/j507/RotatingMHD/blob/68cad27b09555b11308468fbfd5287150eca9634/source/navier_stokes_projection/diffusion_step_methods.cc#L98-L103
allowed to narrow down the error. It seems that the initial residual (`DEAL:GMRES::Starting value`) is much larger when PETSc is used, see log below. This might affect the convergence.

Relevant output of PETSc:
```
Step =      2, Current time = +5.000000e-02, Next time step = +2.500000e-02
    dp = +2.674982e+01 C_d = +5.115431e+01 C_l = -1.540453e-01
  Navier Stokes: Assembling advection matrix... done!
  Navier Stokes: Assembling diffusion step's right hand side... done!
    Right-hand side's L2-norm = +9.336877e+01
  Navier Stokes: Solving the diffusion step...
  Tolerance for the residual: +9.336877e-08
DEAL:GMRES::Starting value 9039.43
DEAL:GMRES::Failure step 2000 value 751.969
```
Relevant output of Trilinos:
```
Step =      2, Current time = +5.000000e-02, Next time step = +2.500000e-02
    dp = +2.674982e+01 C_d = +5.115431e+01 C_l = -1.540452e-01
  Navier Stokes: Assembling advection matrix... done!
  Navier Stokes: Assembling diffusion step's right hand side... done!
    Right-hand side's L2-norm = +9.336877e+01
  Navier Stokes: Solving the diffusion step...
  Tolerance for the residual: +9.336877e-08
DEAL:GMRES::Starting value 524.280
DEAL:GMRES::Convergence step 96 value 7.95184e-08
```
Note that the norm of the right-hand side and the computed tolerance are both equal for PETSc and Trilinos.